### PR TITLE
Hab3 benchmarking scripts

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Adds benchmarking scripts and configs for Habitat v0.3.0 features and tasks.

## How Has This Been Tested

Run locally.

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

**Review by xavierpuigf — APPROVED** [review]
LGTM

File: habitat-lab/habitat/config/benchmark/rearrange/hab3_bench/multi_agent_bench.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# @package _global_
```
Comment (akshararai) [inline habitat-lab/habitat/config/benchmark/rearrange/hab3_bench/multi_agent_bench.yaml:1]: call this `spot_human_bench.yaml`

File: habitat-lab/habitat/config/benchmark/rearrange/hab3_bench/single_agent_bench.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# @package _global_
```
Comment (akshararai) [inline habitat-lab/habitat/config/benchmark/rearrange/hab3_bench/single_agent_bench.yaml:1]: `spot_bench.yaml`

File: scripts/hab3_bench/bench_runner.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+export OMP_NUM_THREADS=2
+export MAGNUM_LOG=quiet
+export HABITAT_SIM_LOG=quiet
+
+# NO_SKINNING="habitat.simulator.agents.agent_1.articulated_agent_urdf='data/hab3_bench_assets/humanoids/female_0/female_0_no_skinning.urdf'"
+
+TASK_SPEC="habitat.task.task_spec=tidy_house_2obj"
+PDDL_DOMAIN_DEF="habitat.task.pddl_domain_def=fp"
+
+# REMOVE_ORACLE="~habitat.task.actions.oracle_nav_action"
```
Comment (akshararai) [inline scripts/hab3_bench/bench_runner.sh:22]: uncomment?

File: scripts/hab3_bench/bench_runner.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    done
+
+    #Single agent robot
+    python scripts/hab3_bench/hab3_benchmark.py --cfg benchmark/rearrange/hab3_bench/spot_oracle.yaml --n-steps "$NUM_STEPS" --n-procs "$j" --out-name "robot_oracle_$i" "$TASK_SPEC" "$PDDL_DOMAIN_DEF" "$DATA_DEFAULT"
+
+
+    # Humanoid oracle
+    python scripts/hab3_bench/hab3_benchmark.py --cfg benchmark/rearrange/hab3_bench/humanoid_oracle.yaml --n-steps "$NUM_STEPS" --n-procs "$j" --out-name "human_oracle_$i" "$TASK_SPEC" "$PDDL_DOMAIN_DEF" "$DATA_DEFAULT"
+
+    # TODO: uncomment soon
```
Comment (akshararai) [inline scripts/hab3_bench/bench_runner.sh:63]: uncomment?

File: scripts/hab3_bench/bench_runner.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    # # Humanoid oracle, no pick no skin
+    # python scripts/hab3_bench/hab3_benchmark.py --cfg benchmark/rearrange/hab3_bench/humanoid_oracle.yaml --n-steps "$NUM_STEPS" --n-procs "$j" --out-name "human_oracle_nopick_noskin_$i" "$TASK_SPEC" "$PDDL_DOMAIN_DEF" "$DATA_DEFAULT" "$REMOVE_PICK" "$NO_SKINNING"
```
Comment (akshararai) [inline scripts/hab3_bench/bench_runner.sh:65]: uncomment?

File: scripts/hab3_bench/h3_plot_bench.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        bbox_inches="tight",
+    )
+    plt.clf()
+
+
+# plot rearrange benchmark numbers
+name_list = [
+    ("robot_human_oracle", "[Robot-Human Oracle Paths]"),
+    ("robots_oracle", "[Robots Oracle]"),
+    # ("robot_human_vel", "[Robot-Human Vel]"),
```
Comment (akshararai) [inline scripts/hab3_bench/h3_plot_bench.py:75]: uncomment or remove

**Review by akshararai — APPROVED** [review]
took a quick glance, overall looks good, except some clean up.
We should test out the readme instructions before 5pm.

File: habitat-lab/habitat/articulated_agents/manipulator.py [hunk 7]
```diff
@@ -303,8 +303,10 @@ def gripper_joint_pos(self):
     def gripper_joint_pos(self, ctrl: List[float]):
         """Kinematically sets the gripper joints and sets the motors to target."""
         joint_positions = self.sim_obj.joint_positions
+        mt = self.sim_obj.motion_type
         for i, jidx in enumerate(self.params.gripper_joints):
-            self._set_motor_pos(jidx, ctrl[i])
+            if mt == MotionType.DYNAMIC:
```
[the remark this answers is not in the document]
Comment (aclegg3) [reply habitat-lab/habitat/articulated_agents/manipulator.py:308]: Yeah, you'll see here that the function is actually to set the joint state kinematically. That is still done when motion type is not dynamic. The motor position settings is an additional step which is only necessary if we have motors.

File: habitat-lab/habitat/config/benchmark/rearrange/hab3_bench/multi_agent_bench.yaml (hunk 1 above)
Comment (aclegg3) [reply habitat-lab/habitat/config/benchmark/rearrange/hab3_bench/multi_agent_bench.yaml:1]: both spot_spot and spot_humanoid use this as the foundation. Maybe it should stay?

**Pull request merged**